src/Search/Extensions/IndexableExtension.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace SilverStripe\FullTextSearch\Search\Extensions;
+
+use SilverStripe\ORM\DataExtension;
+
+/**
+ * Checks if a DataObject is publically viewable thus able to be added or retrieved from a publically searchable index
+ */
+class IndexableExtension extends DataExtension
+{
+    const INDEXABLE_STATE_UNCHECKED = 'INDEXABLE_STATE_UNCHECKED';
+
+    const INDEXABLE_STATE_ALLOWED = 'INDEXABLE_STATE_ALLOWED';
+
+    const INDEXABLE_STATE_DISALLOWED = 'INDEXABLE_STATE_DISALLOWED';
+
+    /**
+     * @var string
+     */
+    protected $indexableState;
+
+    /**
+     * @return bool
+     */
+    public function isIndexable()
+    {
+        if (is_null($this->owner->indexableState)) {
+            $this->owner->indexableState = self::INDEXABLE_STATE_UNCHECKED;
+        }
+
+        if ($this->owner->indexableState !== self::INDEXABLE_STATE_UNCHECKED) {
+            return $this->owner->indexableState === self::INDEXABLE_STATE_ALLOWED;
+        }
+
+        // This will also call $this->owner->getShowInSearch() if it exists
+        if (isset($this->owner->ShowInSearch) && !$this->owner->ShowInSearch) {
+            $this->owner->indexableState = self::INDEXABLE_STATE_DISALLOWED;
+            return false;
+        }
+
+        // TODO: check if class is omitted from canView() check
+        // TODO: Member::actAs(null, ... to test an an anonymous user .. possibly make this a function param
+        // TODO: uncomment - breaks unit tests as is
+//        if (!$this->owner->canView()) {
+//            $this->owner->indexableState = self::INDEXABLE_STATE_DISALLOWED;
+//            return false;
+//        }
```

Comment:
Let's drop this out before merging, I'll reintroduce it in a functioning state in my PR.

Reply:
removed